### PR TITLE
fix: Dynamic duration was always overwritten with the default (30)

### PR DESCRIPTION
**Pull request by emrysal**
## What does this PR do?

multipleDuration useEvent was never set in case of dynamic event - so couldn't be used in the booker store.

### CODE REVIEW

**Review by Udit-takkar — COMMENTED** [review]
@emrysal I am seeing wrong slots when i changed the duration from 30 to 60.


Screenshot 2023-11-22 at 9 40 42 PM

Screenshot 2023-11-22 at 9 42 31 PM

and when I tried to book the slot (10:30Am-11:30AM 60min). I am getting error

Could not book the meeting.
Some of the hosts are unavailable for booking.

http://localhost:3000/pro+free

'pro' has a google calendar connected which has a busy event till 2:00 PM 
'free' doesn't have any calendar connected.

**Review by zomars — CHANGES_REQUESTED** [review]
We got some type check uh-ohs in there

**Review by ThyMinimalDev — DISMISSED** [review]
I believe type errors were not linked to this pr, double checking Udit comment before approving

**Comment by ThyMinimalDev** [comment]
> @emrysal I am seeing wrong slots when i changed the duration from 30 to 60.
> 
> Screenshot 2023-11-22 at 9 40 42 PM Screenshot 2023-11-22 at 9 42 31 PM
> and when I tried to book the slot (10:30Am-11:30AM 60min). I am getting error
> 
> Could not book the meeting. Some of the hosts are unavailable for booking.
> 
> http://localhost:3000/pro+free
> 
> 'pro' has a google calendar connected which has a busy event till 2:00 PM 'free' doesn't have any calendar connected.

not sure to understand what you mean by wrong slots, it seems fine on my end ?

**Comment by Udit-takkar** [comment]
> not sure to understand what you mean by wrong slots, it seems fine on my end ?


Steps to reproduce this bug locally

http://localhost:3000/pro+free

1) Connect google calendar in one of the account (pro )
2) Create a busy event from 8:30AM - 2PM

Screenshot 2023-11-29 at 7 45 03 PM

3) now go to http://localhost:3000/pro+free

When Duration is 30min slots start are 2:30, 3:00, 3:30 .. 

Screenshot 2023-11-29 at 7 45 41 PM


Now change the duration to 60min and now the slots start at 9:30AM, 10:30AM .. and so on

Screenshot 2023-11-29 at 7 46 36 PM


It ignored the busy event in calendar till 2PM.

**Comment by ThyMinimalDev** [comment]
> > not sure to understand what you mean by wrong slots, it seems fine on my end ?
> 
> Steps to reproduce this bug locally
> 
> http://localhost:3000/pro+free
> 
> 1. Connect google calendar in one of the account (pro )
> 2. Create a busy event from 8:30AM - 2PM
> 
> Screenshot 2023-11-29 at 7 45 03 PM
> 3. now go to http://localhost:3000/pro+free
> 
> When Duration is 30min slots start are 2:30, 3:00, 3:30 ..
> 
> Screenshot 2023-11-29 at 7 45 41 PM
> Now change the duration to 60min and now the slots start at 9:30AM, 10:30AM .. and so on
> 
> Screenshot 2023-11-29 at 7 46 36 PM
> It ignored the busy event in calendar till 2PM.

can't reproduce this error on both this branch and main branch, can't you double check ?

**Comment by Udit-takkar** [comment]
I tried again today with a new calendar connection and i am still getting the same slots as the screenshot above.

**Comment by supalarry** [comment]
I also tried reproducing what Udit reported but could not. 30 and 60 minute slots start at the same time after booking event for the pro account on google calendar.

# Before

## 30 minute slots
before-30
before-30-2

## 60 minute slots
before-60

# After

I connected pro to google account and booked from 8:30am to 2pm
Screenshot 2023-12-07 at 10 10 02

## 30 minute slots
Screenshot 2023-12-07 at 10 10 07

## 60 minute slots
Screenshot 2023-12-07 at 10 10 11

File: packages/features/bookings/Booker/types.ts [hunk 1]
```diff
@@ -64,6 +64,10 @@ export interface BookerProps {
    * otherwise, the default value is selected
    */
   duration?: number | null;
+  /**
+   * Configures the selectable options for a multiDuration event type.
+   */
+  durationConfig?: number[];
```
Comment (supalarry) [inline packages/features/bookings/Booker/types.ts:70]: Shall we name it `durationOptions` instead ? I find that word `config` implies a more complex configuration than just a set of possible duration values.

**Review by zomars — APPROVED** [review]
Issues were addressed

**Pull request merged**